code/_onclick/ventcrawl.dm

```diff
@@ -95,9 +95,13 @@ var/list/ventcrawl_machinery = list(/obj/machinery/atmospherics/unary/vent_pump,
 		return
 	visible_message("<span class='danger'>[src] scrambles into [vent_found]!</span>", \
 	"<span class='warning'>You climb into [vent_found].</span>")
-	pick(playsound(src, 'sound/effects/alien_ventpass1.ogg', 35, 1), playsound(src, 'sound/effects/alien_ventpass2.ogg', 35, 1))
+	if(!istype(src,/mob/living/carbon/Xenomorph/Hunter)) //Hunters silently enter/exit vents.
+		pick(playsound(src, 'sound/effects/alien_ventpass1.ogg', 35, 1), playsound(src, 'sound/effects/alien_ventpass2.ogg', 35, 1))
 	forceMove(vent_found)
 	add_ventcrawl(vent_found)
+	if(istype(src,/mob/living/carbon/Xenomorph/Hunter)) //automatically cancel stealth when we enter the vent
```

Comment:
Instead of this check, make something like:

```
/mob/living/carbon/Xenomorph/Hunter/add_ventcrawl(whatever parameters)
	. = ..()
	cancel_stealth()
```